core/server/models/user.js

```diff
@@ -355,16 +365,24 @@ User = ghostBookshelf.Model.extend({
      * Naive user add
      * Hashes the password provided before saving to the database.
      *
-     * @param {object} data
+     * We have to clone the data, because we remove values from this object.
+     * This is not expected from outside!
+     *
+     * @param {object} dataToClone
      * @param {object} options
      * @extends ghostBookshelf.Model.add to manage all aspects of user signup
      * **See:** [ghostBookshelf.Model.add](base.js.html#Add)
      */
-    add: function add(data, options) {
+    add: function add(dataToClone, options) {
         var self = this,
+            data = _.cloneDeep(dataToClone),
             userData = this.filterData(data),
             roles;
 
+        if (!userData.password) {
+            userData.password = utils.uid(50);
```

Comment:
This could be in the defaults on the model? It also makes the line below redundant (it was to convert null/undefined to `''` I believe)